src/port/port.tcl

```diff
@@ -5693,6 +5693,29 @@ if {[catch {mportinit ui_options global_options global_variations} result]} {
     fatal "Failed to initialize MacPorts, $result"
 }
 
+# Change to port directory if requested
+if {[info exists global_options(ports_dir)]} {
+    set dir $global_options(ports_dir)
+    if {[string first "/" $dir] == -1} {
+        set portname $dir
+        if {[catch {mportlookup $portname} result]} {
+            ui_debug $::errorInfo
+            fatal "lookup of portname $portname failed: $result"
+        }
+        if {[llength $result] < 2} {
+            ui_error "port -D failed to look up $portname: no such port"
+            exit 1
+        }
+        array set portinfo [lindex $result 1]
+        set dir $portinfo(portdir)
```

Comment:
I just noticed this is flawed and it only worked by accident in my tests. The `portdir` here is a path relative to the root of the ports tree, but I wanted an absolute path.